### PR TITLE
Feature#3 home desing

### LINKED ISSUE #3: Home Desing
# REQUIREMENTS
As a user, I want to see a screen where it shows me all the complaints made by all users.
A toolbox will be added with which I will be able to navigate in the application, after having started the session.

# DESIGN
image

# Acceptance Criteria
List all required acceptance criteria
- [ ] Show a list with all the complaints saved in the database.
- [ ] Show toolbar.

### PR DESCRIPTION
**Pull request by DiegoSpielzeug**
# Description
I did the layout for HommScreen and the functionality to get the info from the firestore
  
  ## Screenshots 
Screenshot_1621987037

### CODE REVIEW

**Review by jkzasori — DISMISSED** [review]
I think you could improve your design but overall your app it's okay.

File: DenunciaCiudadana/src/assets/styles/styleHomeDesing.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        backgroundColor: "#E50914",
+        padding: 10,
+        borderRadius: 7,
+    },
+    textButton:{
+        color: 'white'
+    }
+})
+
+export { stylesHome }
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/assets/styles/styleHomeDesing.js]: Add new line here

File: DenunciaCiudadana/src/assets/styles/styleHomeDesing.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        marginVertical: 7
+    },
+    buttonContainer:{
+        flexDirection: 'row',
+        justifyContent: 'flex-end',
+        paddingBottom: 5
+    },
+    button:{
+        alignItems: "center",
+        backgroundColor: "#E50914",
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/assets/styles/styleHomeDesing.js]: All the colors, should be on a constant file and you call the color that you need.

Example:
```suggestion
        backgroundColor: colors.white,
```

File: DenunciaCiudadana/src/components/Complaint.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import {View, Text, Image, TouchableOpacity} from 'react-native';
+import {stylesHome} from '../assets/styles/styleHomeDesing';
+
+const Complaint = ({item}) => {
+  return (
+    <View style={stylesHome.complaint}>
+      <View>
+        <Text style={stylesHome.titleComplaint}>Nos cortaron el internet</Text>
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/components/Complaint.js:10]: Try to move all the text into the text constant file, it helps you with the i18n.

```suggestion
        <Text style={stylesHome.titleComplaint}>{texts.someoneDidNotPayBill}</Text>
```

File: DenunciaCiudadana/src/navigation/LogOut.js [hunk 4]
```diff
@@ -4,8 +4,6 @@ import {createStackNavigator} from '@react-navigation/stack';
 import {NavigationContainer} from '@react-navigation/native';
 import styles from '../assets/styles/styles';
 
-import {exampleActions} from '../actions/exampleActions';
-
 const Stack = createStackNavigator();
 
 //provisional screen
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/navigation/LogOut.js:9]: Remove this comment

File: DenunciaCiudadana/src/screens/HomeScreen.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import {useDispatch, useSelector} from 'react-redux';
+import {startLoadingComplaints} from '../actions/complaintsAction';
+
+const HomeScreen = () => {
+  const dispatch = useDispatch();
+  const data = useSelector(state => state.compliants.allComplaints);
+  useEffect(() => {
+    dispatch(startLoadingComplaints());
+  }, []);
+  return (
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/screens/HomeScreen.js:16]: Improve the indentations, be careful with creating readable code

File: DenunciaCiudadana/src/components/Complaint.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        </View>
+        <Text>{item.description}</Text>
+      </View>
+      <View>
+        {item.image ? (
+          <Image style={stylesHome.img} source={{uri: `${item.image}`}} />
+        ) : null}
+        {item.image2 ? (
+          <Image style={stylesHome.img} source={{uri: `${item.image2}`}} />
+        ) : null}
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/components/Complaint.js]:
```suggestion
        {item.image2 ?? (<Image style={stylesHome.img} source={{uri: `${item.image2}`}} />)}
```

If you have a chance read this article.
https://dev.to/ucvdesh/stop-returning-null-components-312k

File: DenunciaCiudadana/src/actions/complaintsAction.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  return async dispatch => {
+    const getComplaints = await firestore().collection('complaints').get();
+    const complaints = await [];
+
+    getComplaints.forEach(doc => {
+      complaints.push({
+        id: doc.id,
+        ...doc.data(),
+      });
+    });
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/actions/complaintsAction.js:14]: Why `forEach`? Did you try with `.map`?

File: DenunciaCiudadana/src/components/Complaint.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import {View, Text, Image, TouchableOpacity} from 'react-native';
+import {stylesHome} from '../assets/styles/styleHomeDesing';
+
+const Complaint = ({item}) => {
+  return (
+    <View style={stylesHome.complaint}>
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/components/Complaint.js]:
```suggestion
const Complaint = ({item}) => {
  const { name, area, description, image, image2 } = item;
 
  return (
    <View style={stylesHome.complaint}>
```

File: DenunciaCiudadana/src/components/Complaint.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+          <Text>{item.name}</Text>
+          <Text>{item.area}</Text>
+        </View>
+        <Text>{item.description}</Text>
+      </View>
+      <View>
+        {item.image ? (
+          <Image style={stylesHome.img} source={{uri: `${item.image}`}} />
+        ) : null}
+        {item.image2 ? (
```
Comment (fernandoamz) [inline DenunciaCiudadana/src/components/Complaint.js]: The name image2 can be better.

```suggestion
        {item.image2 ? (
```

If you have chance I recommend you read this post.

https://dev.to/coly010/clean-javascript-10-tips-37f4

**Review by fernandoamz — CHANGES_REQUESTED** [review]

File: DenunciaCiudadana/src/components/Complaint.js (hunk 3 above)
Comment (DiegoSpielzeug) [reply DenunciaCiudadana/src/components/Complaint.js:10]: It is just for the design, in the next commit I did the change, where I got the info for firebase

File: DenunciaCiudadana/src/navigation/LogOut.js (hunk 4 above)
Comment (DiegoSpielzeug) [reply DenunciaCiudadana/src/navigation/LogOut.js:9]: Same, I removed this in the next commit

File: DenunciaCiudadana/src/actions/complaintsAction.js (hunk 7 above)
Comment (DiegoSpielzeug) [reply DenunciaCiudadana/src/actions/complaintsAction.js:14]: It is the function that appear in the documentation in firebase.com, for make that  kind of operation with data

**Review by jkzasori — APPROVED** [review]

**Review by LeyaimMagma — APPROVED** [review]

**Pull request merged**